packages/shipjs/src/step/release/createGitHubRelease.js

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs';
+import path from 'path';
+import tempWrite from 'temp-write';
+
+import runStep from '../runStep';
+
+function getChangelog(version) {
+  const changelogMatcher = new RegExp(
+    `#+?[\\s\\[]*?(${version})(.|\n)+?(?=#+?[\\s\\[]*?\\d\\.\\d|$)`
+  );
+  const changelogPath = fs.resolve(process.cwd(), 'CHANGELOG.md');
+  try {
+    const changelogFile = fs.readFileSync(changelogPath, 'utf-8').toString();
+    const changelogMatch = changelogFile.match(changelogMatcher);
+    if (changelogMatch !== null) {
+      return changelogMatch[0];
+    }
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      return null;
+    }
+    throw err;
+  }
+  return null;
+}
+
+export default ({ version, config, dir, dryRun }) =>
+  runStep({ title: 'Creating the GitHub Release' }, ({ run }) => {
+    if (config.updateChangelog !== true) return;
+
+    const { getTagName } = config;
+    const tagName = getTagName({ version });
+
+    const existingChangelog = getChangelog(version);
```

Comment:
Let's pass `dir` to `getChangelog` because `dir` can be different from `process.cwd()` when `--dir xxx` is given from command line option.